lib/jsdom/living/events/MouseEvent-impl.js

```diff
@@ -16,31 +16,48 @@ class MouseEventImpl extends UIEventImpl {
   }
   get pageX() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._pageX;
     }
     const offset = wrapperForImpl(this.view)?.scrollX || 0;
     return offset + this.clientX;
   }
   get pageY() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._pageY;
     }
     const offset = wrapperForImpl(this.view)?.scrollY || 0;
     return offset + this.clientY;
   }
   get offsetX() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._offsetX;
     }
     return this.pageX;
   }
   get offsetY() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._offsetY;
     }
     return this.pageY;
   }
 
+  get target() {
+    return super.target;
+  }
+
+  set target(target) {
+    super.target = target;
```

Comment:
Another way to implement this, maybe more elegant, would be a `_receivedTarget` method on the parent `EventImpl` class:
```js
class EventImpl {
  set target(target) {
    this._target = target;
    this._receivedTarget();
  }

  _receivedTarget() {}
}

class MouseEventImpl {
  _receivedTarget() {
    // calculate offsets from this.target
  }
}
```
Chrome's `Event` implementation also has a `::ReceivedTarget` method, and JSDOM also has similar prior art in the `_activationBehavior()` method on elements.
```